awsauth/roundtripper.go

```diff
@@ -0,0 +1,67 @@
+package awsauth
+
+import (
+	"bytes"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	dphttp "github.com/ONSdigital/dp-net/v2/http"
+)
+
+type AwsSignerRoundTripper struct {
+	signer       *Signer
+	roundTripper http.RoundTripper
+}
+
+type Options struct {
+	// InsecureSkipVerify controls whether a client verifies the server's certificate
+	// chain and host name. If InsecureSkipVerify is true, crypto/tls accepts any
+	// certificate presented by the server and any host name in that certificate.
+	// In this mode, TLS is susceptible to machine-in-the-middle attacks unless custom
+	// verification is used. This should be used only for testing or in combination
+	// with VerifyConnection or VerifyPeerCertificate.
+	TlsInsecureSkipVerify bool
+}
+
+var defaultAWSTransport = dphttp.DefaultTransport
+
+func NewAWSSignerRoundTripper(awsFilename, awsProfile, awsRegion, awsService string, options ...Options) (*AwsSignerRoundTripper, error) {
+
+	if awsRegion == "" || awsService == "" {
+		return nil, fmt.Errorf("aws region and service should be valid options")
+	}
+
+	awsSigner, err := NewAwsSigner(awsFilename, awsProfile, awsRegion, awsService)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create aws v4 signer: %w", err)
+	}
+
+	if len(options) > 0 && options[0].TlsInsecureSkipVerify {
```

Comment:
(question): do we actually need to provide ```TlsInsecureSkipVerify``` option to the consumer ? I mean I am not sure whether we should allow any certificate that comes from the server, we should only allow domain specific certificate right ?

Reply:
This is for developer use, to be able to connect from localhost to Elasticsearch cluster

Reply:
in that case i guess we need to control the activation of this using an environment variable right ?

Reply:
yep - that is what i've done on my branch in dp-search-api, and also made a new make target local

Reply:
I could update the docs in awsauth to make it clear this should only be used when testing locally